### DISCUSSION TITLE
Error: export 'XRRigidTransform' (imported as 'XRRigidTransform') was not found in 'three'

### DISCUSSION THREAD

**Discussion by IRobot1** [nartc/angular-three#127] [Q&A] [answered]
If you add the following code to any sandbox project

```ts
        const offsetPosition = { x: 0, y: 0, z: 0, w: 1 };
        const offsetRotation = { x: 0, y: 0, z: 0, w: 1 };
        const transform = new XRRigidTransform(offsetPosition, offsetRotation);
```

ng build shows the following error
![image](https://user-images.githubusercontent.com/25032599/170985285-b2f8341a-e953-4b0e-b17c-ce22e0731456.png)

Can you confirm you get the following error? 
Any help is appreciated

**Comment by IRobot1** [comment]
Browser: [Edge]
NGT [5.4.0]
Three.js version: [r140 or r141]

I understand this is supposed to be part of the browser API like document or window.  However, there definitions can still be located.  Neither Visual Studio or ng build can located the definition for XRRigidTransform.  

It shows up as an import from three, but then complains that it can't be found during build.

I just need a confirmation that I'm not the only one seeing this problem and some suggestions (from your vast knowledge) on some things to try next.

**Comment by IRobot1** [comment]
If I paste this code in a new angular project, I get the same build error, so its nothing related to three or angular-three.  What am I missing?

**Comment by IRobot1** [comment]
Have you had a chance to check the above code works in any of your angular projects?

The following works
```html
<html lang="en">
<body>
    <script type="module">
        let transform = new XRRigidTransform();
        console.warn(transform)
    </script>
</body>
</html>
```
but not when the same two lines of code are added to a new angular project.

**Reply by nartc** [reply]
Install `@types/webxr` then add `webxr` to your `tsconfig`
<img width="427" alt="image" src="https://user-images.githubusercontent.com/25516557/171202792-42bcdadf-8dda-4765-a0c9-980d82f7f391.png">

**Comment by IRobot1** [comment]
I tried this.  It didn't help.  Did this work for you?

**Reply by nartc** [reply]
Yes. 
<img width="584" alt="image" src="https://user-images.githubusercontent.com/25516557/171217010-629cfb91-8cba-4d73-9eca-c2d09e0dd6a5.png">
<img width="1255" alt="image" src="https://user-images.githubusercontent.com/25516557/171217014-72667776-fa68-4a87-9f9f-0f4ba020b998.png">
compiled successful
<img width="678" alt="image" src="https://user-images.githubusercontent.com/25516557/171217037-700a8507-6503-4a60-b186-b672be47f21b.png">
console log

**Comment by IRobot1** [comment]
I used the following work-around for now. I created a Javascript XRUtils file

```js
function RigidTransformXR(args) {
  return new XRRigidTransform(args);
}
```
Updated angular.json to reference file in scripts 
```json
            "scripts": [
              "src/assets/XRUtils.js"
            ]
```
Then in component where I want to call, I added declaration
```ts
declare var RigidTransformXR: any;
...

    const offsetPosition = { x: 0, y: 0, z: 0, w: 1 };
    const offsetRotation = { x: 0, y: 0, z: 0, w: 1 };
    const transform = new RigidTransformXR(offsetPosition, offsetRotation);
```

**Comment by IRobot1** [comment]
can you zip your project and share?

**Reply by nartc** [reply]
Will do when I get back to the laptop

**Comment by IRobot1** [comment, marked as answer]
Apparently adding `types: ["webxr"] to tsconfig doesn't work, but adding to tsconfig.app.json does work.

**Reply by nartc** [reply]
It's how tsconfig works. It only affects the files that the specific tsconfig is including. In an AngularCLI app, `tsconfig.json` only contains the compilation configurations whereas `tsconfig.app.json` tells typescript what the files to compile for the application. This way, when you have like tests with different extensions, or a library, different tsconfig files are pulled in to only compile the correct files
